Documentation/Design/data_flow.svg

```diff
@@ -0,0 +1 @@
ontent: unsafe center; width: 118px; height: 1px; padding-top: 670px; margin-left: 11px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; ">config</div></div></div></foreignObject><text x="70" y="674" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">config</text></switch></g><ellipse cx="340" cy="830" rx="60" ry="40" fill="#e6ffcc" stroke="#000000" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 118px; height: 1px; padding-top: 830px; margin-left: 281px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; ">compile<br />link</div></div></div></foreignObject><text x="340" y="834" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">compile...</text></switch></g><path d="M 160 540 L 210 540 L 240 570 L 240 640 L 160 640 L 160 540 Z" fill="#ffffcc" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><path d="M 210 540 L 210 570 L 240 570 Z" fill-opacity="0.05" fill="#000000" stroke="none" pointer-events="all"/><path d="M 210 540 L 210 570 L 240 570" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 78px; height: 1px; padding-top: 590px; margin-left: 161px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><span>Replace with call analysis?</span></div></div></div></foreignObject><text x="200" y="594" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">Replace with...</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```

Comment:
we'll delete this old diagram, sorry, it seems there's a few extraneous files in this pr!